pydantic/decorators.py

```diff
@@ -83,46 +88,109 @@ def __call__(
     _core_schema.FieldWrapValidatorFunction,
 ]
 
+V1RootValidator = Union[
+    _V1RootValidatorClsMethod,
+    _decorators.V1RootValidatorFunction,
+]
+
+
+# Allow both a V1 (assumed pre=False) or V2 (assumed mode='after') validator
+# We lie to type checkers and say we return the same thing we get
+# but in reality we return a proxy object that _mostly_ behaves like the wrapped thing
+_V1ValidatorType = TypeVar('_V1ValidatorType', bound=Union[V1Validator, 'classmethod[Any]', 'staticmethod[Any]'])
+_V2BeforeAfterOrPlainValidatorType = TypeVar(
+    '_V2BeforeAfterOrPlainValidatorType',
+    bound=Union[V2Validator, 'classmethod[Any]', 'staticmethod[Any]'],
+)
+_V2WrapValidatorType = TypeVar(
+    '_V2WrapValidatorType', bound=Union[V2WrapValidator, 'classmethod[Any]', 'staticmethod[Any]']
+)
+_V1RootValidatorFunctionType = TypeVar(
+    '_V1RootValidatorFunctionType',
+    bound=Union[
+        _decorators.V1RootValidatorFunction,
+        _V1RootValidatorClsMethod,
+        'classmethod[Any]',
+        'staticmethod[Any]',
+    ],
+)
+
 
-@overload
 def validator(
+    __field: str,
     *fields: str,
-    check_fields: bool | None = ...,
-    sub_path: tuple[str | int, ...] | None = ...,
+    pre: bool = False,
+    check_fields: bool | None = None,
     allow_reuse: bool = False,
-) -> Callable[[V2Validator | V1Validator], classmethod[Any]]:
-    ...
+    each_item: bool = False,
+) -> Callable[[_V1ValidatorType], _V1ValidatorType]:
+    """
+    Decorate methods on the class indicating that they should be used to validate fields
+    :param fields: which field(s) the method should be called on
+    :param sub_path: TODO
```

Comment:
I noticed sub_path isn't there, but it's still in this docstring. Should this docstring be updated?

Reply:
thanks removed it